test/LinkerSections/function_sections.swift

```diff
@@ -0,0 +1,10 @@
+// REQUIRES: OS=linux-gnu || OS=linux-androideabi || OS=linux-android || OS=freebsd
```

Comment:
Can we use `%target-object-format` and compare it to `ELF` instead?

Reply:
Not in the `REQUIRES` directive:
```
ValueError: couldn't parse text: '%target-object-format=elf'
in expression: '%target-object-format=elf'
in REQUIRES: directive on test line 1
```

If there is another way to limit it to ELF, that'd be cool.

Reply:
That is rather unfortunate.  Out of curiosity, what happens if you use `-function-sections` on MachO or PE/COFF?  I'm wondering if we should actually prevent that in the driver.  Apple's usage of MachO actually uses `.subsections_via_symbols` which means that everything is atomized and you effectively are doing `-function-sections`, and PE/COFF's symbol resolution is pretty close to this already.

Reply:
I don't have easy access to the other platforms. Also, is the linker / the map file format the same, at this point?

Reply:
The linker map is linker dependent unfortunately.